streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java

```diff
@@ -96,6 +107,14 @@ public SubscriptionInfo(final int version,
             data.setLatestSupportedVersion(latestSupportedVersion);
         }
 
+        if (version >= 8) {
+            if (shutdownRequested == null) { //check is required for testing because the mock sets it to null
```

Comment:
We should instead mock realistic values. Having extra branches in the code just to handle the default behavior of EasyMock creates maintenance problems.